Let's Encrypt update (#1136)
OctopusDeploy/docs: docs/security/exposing-octopus/lets-encrypt-integration.md

At this point, we recommend enabling [Force SSL](/docs/security/exposing-octopus/expose-the-octopus-web-portal-over-https.md#ForcingHTTPS) and [HSTS](/docs/security/exposing-octopus/expose-the-octopus-web-portal-over-https.md#HSTS).

## Let's Encrypt for Containers
The integration for Let's Encrypt was designed for self-hosted Octopus Server installations running on a Windows server. There are currently no plans to support Let's Encrypt when running [Octopus in a container](docs/installation/octopus-in-container/index.md).

> 

Reviewer: ```suggestion

In **Octopus 2020.5** and earlier, when attempting to configure the Let's Encrypt integration on an Octopus Server Linux container, it would fail with a message similar to `We received an error 'Unix LocalMachine X509Stores are read-only for all users`.

```

As Octopus Deploy grows into the Linux and container space, there needs to be several options that customers and users can utilize to ensure they can still use SSL/TLS certifications for the Octopus Deploy platform.

> For Linux and Windows containers moving forward. Let's Encrypt will not be an option in the UI or the API. Instead, you can use many of the standard ways we see today in containers and Kubernetes to enable HTTPS if you are running inside of a container.

Reviewer: ```suggestion
Customers running Octopus in a container that wish to secure the Octopus Portal to be accessible over HTTPS can do so in a number of standard ways, which are discussed in more detail below.
```

### Nginx Proxy

> If you are running an Octopus Deploy container on Docker without an orchestration platform like Kubernetes, you can set up an Nginx reverse proxy with TLS termination. 

Reviewer: ```suggestion
If you are running an Octopus in a Docker container without an orchestration platform like Kubernetes, you can set up an NGINX reverse proxy with TLS termination.
```

> There is a great piece of documentation on the Nginx website for setting up TLS/SSL termination, which you can find [here](https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-http/)

Reviewer: ```suggestion
For more information on configuring TLS/SSL termination in NGINX, refer to the [documentation](https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-http/).
```

### Let's Encrypt and Traefik

> If you'd still like to use Let's Encrypt with Docker and you aren't using an orchestration platform, you can combine Let's Encrypt with Traefik, a popular and cloud native HTTP reverse proxy and load balancer. You can find the documentation for setting up Let's Encrypt and Traefik [here](https://doc.traefik.io/traefik/v1.7/user-guide/docker-and-lets-encrypt/)

Reviewer: ```suggestion
Another popular and cloud-native HTTP reverse proxy and load balancer is [Traefik](https://traefik.io/traefik/). You can combine this with Let's Encrypt to support TLS/SSL termination with your Octopus container. To find out more, refer to  the [Traefik documentation](https://doc.traefik.io/traefik/v1.7/user-guide/.docker-and-lets-encrypt/).
```

> ### Kubernetes and Octopus Deploy

Reviewer: ```suggestion
### Kubernetes and Octopus Deploy {#kubernetes-and-octopus}
```

> If you're going the orchestration and containerization route with say, Kubernetes and Docker, there are a plethora of ingress controllers that can be used. There is an extensive list that you can find [here](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/) that goes over many open-source, extremely popular platforms that include Traefik, HAProxy Ingress, Istio, and many others.

Reviewer: ```suggestion
If you plan to use an orchestrator with your container say, Kubernetes and Docker, an ingress controller can be used to secure the Octopus Portal. You can find an extensive list in the  [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/). It includes multiple open-source, popular platforms such as Traefik, HAProxy Ingress, Istio, and many others.
```

At this point, we recommend enabling [Force SSL](/docs/security/exposing-octopus/expose-the-octopus-web-portal-over-https.md#ForcingHTTPS) and [HSTS](/docs/security/exposing-octopus/expose-the-octopus-web-portal-over-https.md#HSTS).

## Let's Encrypt for Containers
The integration for Let's Encrypt was designed for self-hosted Octopus Server installations running on a Windows server. There are currently no plans to support Let's Encrypt when running [Octopus in a container](docs/installation/octopus-in-container/index.md).

In **Octopus 2020.5** and earlier, when attempting to configure the Let's Encrypt integration on an Octopus Server Linux container, it would fail with a message similar to `We received an error 'Unix LocalMachine X509Stores are read-only for all users`.

From **Octopus 2020.6**, when running Octopus in a container, the Let's Encrypt integration will no longer be available in the Octopus Portal or via the API. 

> Customers running Octopus in a container that wish to secure the Octopus Portal to be accessible over HTTPS can do so in a number of standard ways, which are discussed in more detail below.

Reviewer: ```suggestion
Customers running Octopus in a container that wish to secure the Octopus Portal to be accessible over HTTPS can do so in a number of standard ways, which are discussed in more detail here.
```
